app/Jobs/DispatchPollJobs.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+namespace App\Jobs;
+
+use App\Models\Device;
+use Illuminate\Bus\Queueable;
+use Illuminate\Contracts\Queue\ShouldQueue;
+use Illuminate\Foundation\Bus\Dispatchable;
+use Illuminate\Queue\InteractsWithQueue;
+use Illuminate\Queue\SerializesModels;
+use Illuminate\Support\Facades\Log;
+use LibreNMS\Config;
+
+class DispatchPollJobs implements ShouldQueue
+{
+    use Dispatchable, InteractsWithQueue, Queueable, SerializesModels;
+
+    /**
+     * @param  string  $device_spec
+     * @param  int  $verbosity
+     * @param  array<string, bool|string[]>  $module_overrides
+     */
+    public function __construct(
+        public string $device_spec,
+        public int $verbosity = -1,
+        public array $module_overrides = [],
+    ) {
+    }
+
+    /**
+     * Execute the job.
+     */
+    public function handle()
+    {
+        // Make sure we have configured job queueing
+        if (\config('queue.default') != 'sync') {
+            // Queue poller jobs if not configured for service worker
+            if (! Config::get('service_poller_enabled')) {
+                $devices = Device::whereDeviceSpec($this->device_spec)->select('device_id', 'poller_group')->get();
```

Comment:
This query suffers from Eloquent model initialization as I mentioned earlier.  It is also dispatching work for disabled devices.  This query would be more efficient:

```php
        $devices = DB::table('devices')
            ->select(['device_id', 'poller_group'])
            ->where('disabled', 0)
            ->where(function (Builder $query) {
                $query->whereNull('last_polled')
                    ->orWhereRaw('`last_polled` <= DATE_ADD(DATE_ADD(NOW(), INTERVAL -? SECOND), INTERVAL COALESCE(`last_polled_timetaken`, 0) SECOND)', [$pollfreq]);
            })
            ->orderBy('last_polled_timetaken', 'desc')
            ->get();
```

Reply:
Will do